api/client/client.go

```diff
@@ -2094,7 +2094,7 @@ func (c *Client) CreateAuthenticateChallenge(ctx context.Context, in *proto.Crea
 }
 
 // CreatePrivilegeToken is implemented by AuthService.CreatePrivilegeToken.
-func (c *Client) CreatePrivilegeToken(ctx context.Context, req *proto.CreatePrivilegeTokenRequest) (types.UserToken, error) {
+func (c *Client) CreatePrivilegeToken(ctx context.Context, req *proto.CreatePrivilegeTokenRequest) (*types.UserTokenV3, error) {
```

Comment:
I just noticed that if we ever +1 the version e.g `UserTokenV4`, we have more places to change the signature vs. just updating in `grpcserver.go` and in https://github.com/gravitational/teleport/blob/master/api/types/usertoken.go#L50 (and receiver funcs)

Does this mean that I should be wrapping the token in a `CreatePrivilegeTokenResponse`?

Reply:
I don't mind that we have to touch more places when we bump the token version; one could even argue that it's a good thing, as it draws attention to where and how tokens are used. If that is too hard to do, then it highlights an issue in the solutions we've chosen.

In Go it's usual to return concrete types and leave push interface use to the "edges", only where absolutely needed (and having the interfaces as small as possible).